pipeline/terraform/stack/elastic_pipeline_storage.tf

```diff
@@ -0,0 +1,124 @@
+locals {
+  es_memory = var.is_reindexing ? "29g" : "8g"
+}
+
+resource "ec_deployment" "pipeline_storage" {
+  name = "catalogue-pipeline-storage-${var.pipeline_date}abcd"
+
+  region                 = "eu-west-1"
+  version                = "7.10.2"
+  deployment_template_id = "aws-io-optimized-v2"
+
+  traffic_filter = [
+    var.traffic_filter_platform_vpce_id,
+    var.traffic_filter_public_internet_id,
+  ]
+
+  elasticsearch {
+    topology {
+      zone_count = 1
+      size       = local.es_memory
+    }
+  }
+
+  kibana {
+    topology {
+      zone_count = 1
+      size       = "1g"
+    }
+  }
+}
+
+# We create the username/password secrets in Terraform, and set the values in the
+# Python script.  This ensures the secrets will be properly cleaned up when we delete
+# a pipeline.
+resource "aws_secretsmanager_secret" "es_username" {
+  for_each = toset(concat(local.pipeline_storage_service_list, ["image_ingestor", "work_ingestor", "read_only"]))
+
+  name = "elasticsearch/pipeline_storage_${var.pipeline_date}/${each.key}/es_username"
+}
+
+resource "aws_secretsmanager_secret" "es_password" {
+  for_each = toset(concat(local.pipeline_storage_service_list, ["image_ingestor", "work_ingestor", "read_only"]))
+
+  name = "elasticsearch/pipeline_storage_${var.pipeline_date}/${each.key}/es_password"
+}
+
+# We can't attach the provisioner directly to the Elastic Cloud resource (I'm not
+# entirely sure why), so instead we create a null resource that will be recreated
+# whenever the cluster is created.
+#
+# The local-exec provisioner on this resource runs a script that sets up the
+# Elasticsearch users for the pipeline.
+#
+# Note: this must run *after* the cluster and secrets are created, or the script
+# won't work.
+resource "null_resource" "elasticsearch_users" {
+  triggers = {
+    pipeline_storage_elastic_id = local.pipeline_storage_elastic_id
+  }
+
+  depends_on = [
+    module.pipeline_storage_secrets,
+    aws_secretsmanager_secret.es_username,
+    aws_secretsmanager_secret.es_password,
+  ]
+
+  provisioner "local-exec" {
+    command = "python3 scripts/create_pipeline_storage_users.py ${var.pipeline_date}"
```

Comment:
Fancy!

Reply:
What happens if this fails?

Reply:
If a local-exec provisioner fails, Terraform marks the resource as tainted and tries to recreate it on the next plan/apply cycle.